salt/modules/ini_manage.py

```diff
@@ -397,7 +398,7 @@ def flush(self):
             with salt.utils.files.fopen(self.name, 'w') as outfile:
                 ini_gen = self.gen_ini()
                 next(ini_gen)
-                outfile.writelines(ini_gen)
+                outfile.writelines(salt.utils.stringutils.to_str(ini_gen))
```

Comment:
`ini_gen` is a list here, not a string. Like above, either we need to run `salt.utils.stringutils.to_str()` in a list comprehension, or you can open this file with mode `wb` and use:

```python
outfile.writelines(salt.utils.data.encode(ini_gen))
```